backend/services/implementations/lessonService.ts

```diff
@@ -0,0 +1,61 @@
+import { v4 as uuidv4 } from "uuid";
+
+import MgLesson, { Lesson } from "../../models/lesson.model";
+import { ILessonService } from "../interfaces/lessonService";
+import {
+  LessonDTO,
+  CreateLessonDTO,
+} from "../../types";
+import { getErrorMessage } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+
+const Logger = logger(__filename);
+
+class LessonService implements ILessonService {
+
+  /* eslint-disable class-methods-use-this */
+  async getLessonById(id: string): Promise<LessonDTO> {
+    let lesson: Lesson | null;
+    try {
+      lesson = await MgLesson.findById(id);
+      if (!lesson) {
+        throw new Error(`Lesson id ${id} not found`);
+      }
+    } catch (error: unknown) {
+      Logger.error(`Failed to get entity. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+
+    return {
```

Comment:
I'm honestly not sure if there's any benefit in defining a new object with `return {...` to copy over all the properties of lesson instead of just doing `return lesson`? Not a huge thing.